// x-chalk/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.log = exports.white = exports.bgRed = exports.grey = exports.gray = exports.cyan = exports.magenta = exports.green = exports.yellow = exports.blue = exports.red = exports.c = void 0;
var chalk_1 = __importDefault(require("chalk"));
exports.c = chalk_1.default;
function red() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.red.apply(chalk_1.default, args));
}
exports.red = red;
red.dim = function () {
    var _a;
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log((_a = chalk_1.default.red).dim.apply(_a, args));
};
function blue() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.blue.apply(chalk_1.default, args));
}
exports.blue = blue;
blue.dim = function () {
    var _a;
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log((_a = chalk_1.default.blue).dim.apply(_a, args));
};
function yellow() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.yellow.apply(chalk_1.default, args));
}
exports.yellow = yellow;
yellow.dim = function () {
    var _a;
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log((_a = chalk_1.default.yellow).dim.apply(_a, args));
};
function green() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.green.apply(chalk_1.default, args));
}
exports.green = green;
green.dim = function () {
    var _a;
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log((_a = chalk_1.default.green).dim.apply(_a, args));
};
function magenta() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.magenta.apply(chalk_1.default, args));
}
exports.magenta = magenta;
magenta.dim = function () {
    var _a;
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log((_a = chalk_1.default.magenta).dim.apply(_a, args));
};
function cyan() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.cyan.apply(chalk_1.default, args));
}
exports.cyan = cyan;
cyan.dim = function () {
    var _a;
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log((_a = chalk_1.default.cyan).dim.apply(_a, args));
};
function gray() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.gray.apply(chalk_1.default, args));
}
exports.gray = gray;
gray.dim = function () {
    var _a;
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log((_a = chalk_1.default.gray).dim.apply(_a, args));
};
function grey() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.gray.apply(chalk_1.default, args));
}
exports.grey = grey;
grey.dim = function () {
    var _a;
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log((_a = chalk_1.default.gray).dim.apply(_a, args));
};
function bgRed() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.bgRed.apply(chalk_1.default, args));
}
exports.bgRed = bgRed;
function white() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log.apply(console, args);
}
exports.white = white;
white.dim = function () {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.dim.apply(chalk_1.default, args));
};
function log() {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log.apply(console, args);
}
exports.log = log;
log.dim = function () {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    console.log(chalk_1.default.dim.apply(chalk_1.default, args));
};
